packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -259,6 +290,46 @@ class DismissUpdateDetails {
   final double progress;
 }
 
+/// Details for [TriggerDismissCallback].
+///
+/// See also:
+///
+///   * [Dismissible.shouldTriggerDismiss], which receives this information.
+class TriggerDismissDetails {
+  /// Create a new instance of [TriggerDismissDetails].
+  const TriggerDismissDetails({
+    this.direction = DismissDirection.horizontal,
+    this.reached = false,
+    this.progress = 0.0,
+    this.velocity = 0.0,
+    this.dragExtent = 0.0,
+  });
+
+  /// The direction that the dismissible is being dragged.
+  final DismissDirection direction;
+
+  /// Whether the dismiss threshold is currently reached.
+  final bool reached;
+
+  /// The offset ratio of the dismissible in its parent container.
+  ///
+  /// A value of 0.0 represents the normal position and 1.0 means the child is
+  /// completely outside its parent.
+  ///
+  /// This can be used to synchronize other elements to what the dismissible is doing on screen,
+  /// e.g. using this value to set the opacity thereby fading dismissible as it's dragged offscreen.
+  final double progress;
+
+  /// Distance from the start of the drag to the current position
+  ///
+  /// The drag distance is represented in logical pixels.
+  final double dragExtent;
+
+  /// The component of the velocity parallel to the [DismissDirection].
+  /// A fling away from the center will have a positive value.
+  final double velocity;
```

Comment:
tiny whitespace/wording nit:

```suggestion
  /// Distance from the start of the drag to the current position,
  /// represented in logical pixels.
  final double dragExtent;

  /// The component of the velocity parallel to the [DismissDirection].
  ///
  /// A fling away from the center will have a positive value.
  final double velocity;
```